src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/FileSystemController.cs

```diff
@@ -0,0 +1,294 @@
+        ///   A path to an existing file, such as "QmXarR6rgkQ2fDSHjSY5nM2kuCXKYGViky5nohtwgF65Ec/about"
+        ///   or "QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V"
+        /// </param>
+        [HttpGet, HttpPost, Route("file/ls")]
+        public async Task<FileSystemDetailsDto> Stat(string arg)
+        {
+            var node = await IpfsCore.UnixFsApi.ListFileAsync(arg, Cancel);
+            var dto = new FileSystemDetailsDto
+            {
+                Arguments = new Dictionary<string, string>(),
+                Objects = new Dictionary<string, FileSystemDetailDto>()
+            };
+            dto.Arguments[arg] = node.Id;
+            dto.Objects[node.Id] = new FileSystemDetailDto
+            {
+                Hash = node.Id,
+                Size = node.Size,
+                Type = node.IsDirectory ? "Directory" : "File",
+                Links = node.Links
+                   .Select(link => new FileSystemLinkDto
+                    {
+                        Hash = link.Id,
+                        Name = link.Name,
+                        Size = link.Size
+                    })
+                   .ToArray()
+            };
+            return dto;
+        }
+
+        /// <summary>
+        ///   Add a file.
+        /// </summary>
+        [HttpGet, HttpPost, Route("add")]
+        public async Task Add(IFormFile file,
+            string hash = MultiHash.DefaultAlgorithmName,
+            [ModelBinder(Name = "cid-base")] string cidBase = MultiBase.DefaultAlgorithmName,
+            [ModelBinder(Name = "only-hash")] bool onlyHash = false,
+            string chunker = null,
+            bool pin = false,
+            [ModelBinder(Name = "raw-leaves")] bool rawLeaves = false,
+            bool trickle = false,
+            [ModelBinder(Name = "wrap-with-directory")]
+            bool wrap = false,
+            string protect = null,
+            bool progress = true)
+        {
+            if (file == null)
+                throw new ArgumentNullException(nameof(file));
+
+            var options = new AddFileOptions
+            {
+                Encoding = cidBase,
+                Hash = hash,
+                OnlyHash = onlyHash,
+                Pin = pin,
+                RawLeaves = rawLeaves,
+                Trickle = trickle,
+                Wrap = wrap,
+                ProtectionKey = protect,
+            };
+            if (chunker != null)
+            {
+                if (chunker.StartsWith("size-"))
+                {
+                    options.ChunkSize = int.Parse(chunker.Substring(5), CultureInfo.InvariantCulture);
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException(nameof(chunker));
+                }
+            }
+
+            if (progress)
+            {
+                options.Progress = new Progress<TransferProgress>(StreamJson);
+            }
+
+            // TODO: Accept multiple files.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)